src/ripple/app/paths/impl/DirectStep.cpp

```diff
@@ -43,10 +165,10 @@ class DirectStepI : public StepImp<IOUAmount, IOUAmount, DirectStepI>
     // Charge transfer fees when the prev step redeems
     Step const* const prevStep_ = nullptr;
 
-    // When offerCrossing then ignore trust line quality.
-    bool const offerCrossing_;
+    // Policy that distinguishes behavior between payments and offer crossing.
+    Policy policy_;
+    friend Policy;
```

Comment:
`Policy` is feeling more like it should be a subclass.

The goal of this commit is to isolate the differences between payments and offer crossing into their own entities. the `Policy` implementation does that, and it's a win. We can use inheritance to implement the same thing (and CRTP to avoid virtual function callls). Now that you've implemented a policy-based design, do you still favor this over inheritance?